media/admin/com_virtuemart/controllers/zasilkovna.php

```diff
@@ -295,4 +290,49 @@ private function getExportOrders()
     {
         return isset($_POST['exportOrders']) && is_array($_POST['exportOrders']) ? $_POST['exportOrders'] : [];
     }
+
+    /**
+     * @param array<string, mixed> $data
+     * @return FlashMessage
+     */
+    private function updateZasilkovnaConfig(array $data): FlashMessage
+    {
+        $configStorage = new VirtueMartModelZasilkovna\SessionStorage(JFactory::getSession(), 'packeteryConfig');
+        $configStorage->set(self::FROM_POST, self::FORM_VALUES, $data);
+
+        /** @var VirtueMartModelZasilkovna $model */
+        $model = VmModel::getModel('zasilkovna');
+        $currentData = $model->loadConfig();
+
+        $formValidator = new VirtueMartModelZasilkovna\ConfigurationValidator($data);
+        $formValidator->validate();
+
+        if (!$formValidator->isValid()) {
+            $errors = $formValidator->getErrors();
+            $messages = [];
+
+            //$error is either error string translation key, or array where 1st element is sprintf template and others are sprintf arguments (all untranslated)
+            foreach ($errors as $formField => $error) {
+                if (!is_array($error)) {
```

Comment:
Proč je tohle lepší než volat JText::_() už ve validátoru a vracet jen stringy?
Pokud to jde, tak vracej rovnou hotové stringy.

Reply:
Za mě v tom, že takhle ten náš nový krásný, čistý validátor nemá žádnou závislost na joomlých věcech a tady v tom kontroleru už ta závislost je.

Reply:
Obecně souhlasím, že nemít závislosti na joomlích věcech je super, ale ne za cenu toho, že bychom pak měli takovýhle kód.